SofaKernel/modules/SofaHelper/src/sofa/helper/AdvancedTimer.cpp

```diff
@@ -1406,8 +1406,27 @@ std::map<AdvancedTimer::IdStep, StepData> AdvancedTimer::getStepData(IdTimer id,
 helper::vector<Record> AdvancedTimer::getRecords(IdTimer id)
 {
     TimerData& data = timers[id];
-    for (unsigned int i=0; i<data.records.size(); ++i)
-        data.records[i].label = std::string(AdvancedTimer::IdStep(data.records[i].id));
+    for (Record & r : data.records) {
+        switch (r.type) {
+            case Record::RBEGIN: // Timer begins
+            case Record::REND: // Timer ends
+                r.label = (std::string) AdvancedTimer::IdTimer(r.id);
+                break;
+            case Record::RSTEP_BEGIN: // Step begins
+            case Record::RSTEP_END: // Step ends
+            case Record::RSTEP: // Step
+                r.label = (std::string) AdvancedTimer::IdStep(r.id);
+                break;
+            case Record::RVAL_SET: // Sets a value
+            case Record::RVAL_ADD: // Adds a value
+                r.label = (std::string) AdvancedTimer::IdVal(r.id);
+                break;
+            default:
+                r.label = (std::string) AdvancedTimer::IdObj(r.id);
```

Comment:
should not be IdStep by default? (just compared to the old version)

Reply:
Now that I think about it, the default should never happen. If it does, it means we added a new Id type and we did not treat it in this function.

Casting it to a default id (IdStep or IdObj) will produce undefined behaviour since it will try to get the name value of the id from another factory of id. This will be hard to debug.

For example, if we have:
```
IdTimer         |        IdStep
----------------|-----------------------
1: "Timer1"     |       1: "Step 1"
2: "Timer2"     |       2: "Step 2"
```

Then
```
IdTimer id (1); //  "Timer 1"
std::cout << (std::string) IdStep(id) << std::endl;
//  "Step 1"
```
is wrong.

I suggest we use a msg_error for the default, so that we can easily fix it once we see the error.

What do you think?